tests/pdb_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package tests_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	expect "github.com/google/goexpect"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libvmops"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Pod Disruption Budget (PDB)", decorators.SigCompute, func() {
+
+	It("should ensure the Pod Disruption Budget (PDB) is deleted when the VM is stopped via guest OS", func() {
+		By("Creating test VM")
+		vm := libvmi.NewVirtualMachine(
+			libvmifact.NewCirros(
+				libvmi.WithEvictionStrategy(v1.EvictionStrategyLiveMigrate),
+				libvmi.WithResourceMemory("128Mi"),
+			),
+			libvmi.WithRunStrategy(v1.RunStrategyManual),
+		)
+
+		vm, err := kubevirt.Client().VirtualMachine(testsuite.NamespaceTestDefault).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		libvmops.StartVirtualMachine(vm)
+
+		By("Waiting for the VM to be ready")
+		Eventually(ThisVMIWith(vm.Namespace, vm.Name), 180*time.Second, 1*time.Second).Should(BeAssignableToTypeOf(&v1.VirtualMachineInstance{}))
```

Comment:
`Exist()` instead of `BeAssignableToTypeOf(&v1.VirtualMachineInstance{})`

Reply:
fixed